assets/js/translator.js

```diff
@@ -730,6 +744,131 @@ function translateDoc() {
+
+function translateLink(href) { // eslint-disable-line no-unused-vars
+    $('#webpage').val(href);
+    if(window.history) {
+        // used by window.onpopstate:
+        window.history.pushState({}, document.title, window.location.href);
+    }
+    translateWebpage();
+}
+
+function cleanPage(html) {
+    // Pages like
+    // eslint-disable-next-line max-len
+    // http://www.lapinkansa.fi/sagat/romssa-sami-searvvi-jodiheaddji-daruiduhttinpolitihkka-vaikkuha-ain-olbmuid-guottuide-samiid-birra-15843633/
+    // insert noise using document.write that 1. makes things
+    // enormously slow, and 2. completely mess up styling so e.g. you
+    // have to scroll through a full screen of whitespace before
+    // reaching content. This might mess things up some places – needs
+    // testing – but on the other hand most uses of document.write are
+    // evil.
+    return html.replace(/document[.]write[(]/g,
+        'console.log("document.write "+');
+}
+
+function translateWebpage() {
+    persistChoices('translator', true);
+    if(!$('div#translateWebpage').is(':visible')) {
+        showTranslateWebpageInterface($('#originalText').val().trim());
+    }
+
+    if(pairs[curSrcLang] && pairs[curSrcLang].indexOf(curDstLang) !== -1) {
+        sendEvent('translator', 'translateWebpage', curSrcLang + '-' + curDstLang);
+        $('iframe#translatedWebpage').animate({'opacity': 0.75}, 'fast');
+        $.ajax({
+            url: config.APY_URL + '/translatePage',
+            dataType: 'json',
+            beforeSend: ajaxSend,
+            complete: function () {
+                ajaxComplete();
+                synchronizeTextareaHeights();
+                textTranslateRequest = undefined;
+                $('iframe#translatedWebpage').animate({'opacity': 1}, 'fast');
+            },
+            data: {
+                'langpair': curSrcLang + '|' + curDstLang,
+                'markUnknown': 'no', // TODO: checkbox; also perhaps only remove the #-marks, not *
+                'url': $('input#webpage').val()
+            },
+            success: function (data) {
+                if(data.responseStatus === HTTP_OK_CODE) {
+                    var iframe = $('<iframe id="translatedWebpage" class="translatedWebpage" frameborder="0"></iframe>')[0];
+                    $('#translatedWebpage').replaceWith(iframe);
+                    iframe.contentWindow.document.open();
+                    var html = cleanPage(data.responseData.translatedText);
+                    iframe.contentWindow.document.write(html);
+                    iframe.contentWindow.document.close();
+                    var contents = $(iframe).contents();
+                    contents.find('head')
+                        .append($('<base>').attr('href', $('input#webpage').val()));
+                    $(iframe).load(function () {
+                        contents.find('a')
+                            .map(function (index, a) { // eslint-disable-line array-callback-return
```

Comment:
Um, I understood the difference between usage of `forEach` (would help as it would not return any value) and `map` and also referred to this [stackoverflow link](https://stackoverflow.com/questions/34426458/js-learning-difference-between-foreach-and-map) for the same. Um, as I have an array on which when I replace my `.map()` with `.forEach()`, I am getting an 'Uncaught type error: contents.find(...).forEach is not a function'. I have been trying on this but no luck. Could you help a bit on this? Thanks!